### DISCUSSION TITLE
Consider not minting the NFT when the stream is non-transferable

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#819] [Ideas]
## Problem

Owners of non-transferable streams are confused. See this feedback from a $5IRE airstream recipient:

<img width="400" alt="SCR-20240208-lttd" src="https://github.com/sablier-labs/v2-core/assets/8782666/51de157a-530a-4a9d-ae9b-0eec644be7de">

<img width="400" alt="SCR-20240208-mfxg" src="https://github.com/sablier-labs/v2-core/assets/8782666/e63279d7-0020-4078-97e1-17d348a28be8">

## Proposal

Do not mint an NFT when the user wishes to make it non-transferable.

### Benefits

- No more redundant listings on OpenSea
- No need to implement [this task](https://github.com/sablier-labs/v2-core/issues/818) in the NFT descriptor
- Gas optimization (related: https://github.com/sablier-labs/company-discussions/discussions/36)

### Cons

- Losing the aesthetic benefits (e.g., users valuing the NFT for how it looks)
- Bespoke logic for storing the recipient address
- Additional checks required whenever there are interactions with the NFT featuers, e.g. transfer, burn
- More complicated logic in the UI

## RFC

Cc @sablier-labs/everybody.

**Comment by andreivladbrg** [comment]
> No more redundant listings on OpenSea

What do you mean? We are not listing each NFT, but we are listing a collection. So the contracts still need to be listed on OpenSea; we are not going to have only non-transferable streams.

> No need to implement https://github.com/sablier-labs/v2-core/issues/818 in the NFT descriptor

It's a minor task; it would take us a few minutes.

> Gas optimization

I'm not so sure; it would require additional logic in the `create` function, and the recipient would still need to be stored somewhere in storage. So, if there would be any gas benefits, they would be negligible.

> Cons
Losing the aesthetic benefits (e.g., users valuing the NFT for how it looks)
Bespoke logic for storing the recipient address
Additional checks required whenever there are interactions with the NFT featuers, e.g. transfer, burn
More complicated logic in the UI

These cons simply far outweigh the pros.

The solution is to update the UI when the stream is non-transferable to add something like, "If the stream is transferable, transferring it would cause the rights..."

And to address the task https://github.com/sablier-labs/v2-core/issues/818

**Reply by PaulRBerg** [reply]
> We are not listing each NFT, but we are listing a collection

I was referring to users listing NFTs for sale. See https://github.com/sablier-labs/v2-core/issues/818.

> if there would be any gas benefits, they would be negligible.

You may be right.

> These cons simply far outweigh the pros.

Soft agree

> The solution is to update the UI when the stream is non-transferable to add something like

Yes: https://github.com/sablier-labs/v2-interfaces/issues/874
